### DISCUSSION TITLE
Next.js/Jest failure to recognize automatic React import to tsx files.

### DISCUSSION THREAD

**Discussion by jf-t** [vercel/next.js#18490] [Help] [answered]
# Feature request

## Is your feature request related to a problem? Please describe.

I have a Next.js app that we have been contributing towards for a little over a year now. We were always in a rush to get the thing out the door and now after so long I have decided to implement a Jest/React-testing-library testing framework into the application. Everything was set up fine but I noticed that Jest seems to be having trouble that I am not directly importing React into the `.tsx` files, but rather am just using `import { FunctionComponent } from "react";`. This has worked ever since we started development on the site, since Next.js is automatically importing React as a variable into all `.tsx` files, but it seems like Jest is having trouble picking this up. I am seeing a long line of:
```
'React' refers to a UMD global, but the current file is a module. Consider adding an import instead.
``` 
Highlighting every component used in the tsx.

## Describe the solution you'd like

I would like to:
- Know whether this has been solved or not, I have spent about an hour searching through past issues but I have not found anybody come to a consensus about whether this is an issue or whether this is just something I am going to have to add to every component on the site (we have 362 `.tsx` files as of now...).
- If this is an issue that currently has no solution, then to figure out whether it could be added so that this does not become an issue for anybody else down the line. It is never nice to have to change every file of code that you have written just so that we can add thorough testing.

## Describe alternatives you've considered

The alternative solution is to add `import React` on every `.tsx` file in the site, which isn't necessarily a make-or-break change, but there is an automatic importing in place for the site itself for a reason and it would be nice to see the ideal solution if possible.

## Additional context

I apologize in advance if this has been asked before, but I really did try to search through the backlog of issues.

**Comment by jf-t** [comment]
Anybody have an idea? This is seriously going to hurt the productivity of our testing suite int he upcoming months - I am sure I am not the only one who has experienced this

**Comment by lfades** [comment, marked as answer]
@jf-t Please follow the setup in our [with-jest example](https://github.com/vercel/next.js/tree/canary/examples/with-jest). We already have multiple examples using Jest without an issue, what you're describing is that your Jest setup doesn't seem tobe using the `next/babel` preset (which the example uses). If after checking the example it still doesn't work for you, please provide a reproduction that we can take a look at 🙏

**Reply by pavlovoloshchukbeetroot** [reply]
Nice one. But this will not work with TS.
